Content.Shared/Medical/Cryogenics/SharedCryoPodSystem.cs

```diff
@@ -0,0 +1,185 @@
+        }
+    }
+
+    public void InsertBody(EntityUid uid, EntityUid target, SharedCryoPodComponent cryoPodComponent)
+    {
+        if (cryoPodComponent.BodyContainer.ContainedEntity != null)
+            return;
+
+        if (!HasComp<MobStateComponent>(target))
+            return;
+
+        var xform = Transform(target);
+        cryoPodComponent.BodyContainer.Insert(target, transform: xform);
+
+        var comp = EnsureComp<InsideCryoPodComponent>(target);
+        comp.Holder = cryoPodComponent.Owner;
+        _standingStateSystem.Stand(target, force: true); // Force-stand the mob so that the cryo pod sprite overlays it fully
+
+        UpdateAppearance(uid, cryoPodComponent);
+    }
+
+    public void TryEjectBody(EntityUid uid, EntityUid userId, SharedCryoPodComponent? cryoPodComponent)
+    {
+        if (!Resolve(uid, ref cryoPodComponent))
+        {
+            return;
+        }
+
+        if (cryoPodComponent.Locked)
+        {
+            _popupSystem.PopupEntity(Loc.GetString("cryo-pod-locked"), uid, Filter.Entities(userId));
+            return;
+        }
+
+        EjectBody(uid, cryoPodComponent);
+    }
+
+    public virtual void EjectBody(EntityUid uid, SharedCryoPodComponent? cryoPodComponent)
+    {
+        if (!Resolve(uid, ref cryoPodComponent))
+            return;
+
+        if (cryoPodComponent.BodyContainer.ContainedEntity is not {Valid: true} contained)
+            return;
+
+        cryoPodComponent.BodyContainer.Remove(contained);
+        RemComp<InsideCryoPodComponent>(contained);
+
+        // Restore the correct position of the patient. Checking the components manually feels hacky, but I did not find a better way for now.
+        if (HasComp<KnockedDownComponent>(contained) || _mobStateSystem.IsIncapacitated(contained))
+        {
+            _standingStateSystem.Down(contained);
+        }
+        else
+        {
+            _standingStateSystem.Stand(contained);
+        }
+
+        UpdateAppearance(uid, cryoPodComponent);
+    }
+
+    private void AddAlternativeVerbs(EntityUid uid, SharedCryoPodComponent cryoPodComponent, GetVerbsEvent<AlternativeVerb> args)
+    {
+        if (!args.CanAccess || !args.CanInteract)
+            return;
+
+        // Eject verb
+        if (cryoPodComponent.BodyContainer.ContainedEntity != null)
+        {
+            args.Verbs.Add(new AlternativeVerb
+            {
+                Text = Loc.GetString("cryo-pod-verb-noun-occupant"),
+                Category = VerbCategory.Eject,
+                Priority = 1, // Promote to top to make ejecting the ALT-click action
+                Act = () => TryEjectBody(uid, args.User, cryoPodComponent)
+            });
+        }
+    }
+
+    private void OnEmagged(EntityUid uid, SharedCryoPodComponent? cryoPodComponent, GotEmaggedEvent args)
+    {
+        if (!Resolve(uid, ref cryoPodComponent))
+        {
+            return;
+        }
+
+        cryoPodComponent.PermaLocked = true;
```

Comment:
I'm still highly suss on emag permanently locking someone in considering there's nothing they can personally do about it, I think it should just be a regular lock.

Reply:
Given the patient is no longer put to sleep, what if people could break cryotubes (they are just fancy glass tubes after all) which would let someone trapped inside break out. It could be pretty strong so just using fists might take many hits.

Reply:
My point of view is that for this to happen, you need to consume an emag's charge, so the effect needs to be meaningful. If the patient can just as easily get himself outside the pod, I doubt anyone would waste a charge for this. But from the patient's point of view, they still can scream and speak on the radio, which is useful considering those pods would most of the time be located in the middle of medbay. 
Since you're not asleep, you can actually get yourself outside the pod if you have a crowbar. Crowbar interaction works inside the pod too. I didn't plan on allowing this initially but I guess it can be left in to give a better survival chance to the patient.
And, as @theashtronaut said, you can already break the pod to eject someone stuck in it, but currently if you're inside the pod you cannot attack the pod itself.